src/utils/compute.ts

```diff
@@ -203,7 +200,20 @@ async function getJobs(
     }
   }
 
-  return computeJobs
+  const uniqueAgreementIds = new Set()
+  const uniqueComputeJobs = computeJobs
+    .filter((e) => {
+      const isDuplicate = !uniqueAgreementIds.has(e.agreementId)
+      uniqueAgreementIds.add(e.agreementId)
+      return isDuplicate
+    })
+    .sort(
+      (a, b) =>
+        new Date(Number(b.dateCreated) * 1000).getTime() -
+        new Date(Number(a.dateCreated) * 1000).getTime()
+    )
+
+  return uniqueComputeJobs
```

Comment:
suggestion(non-blocking): refactor for readability - something like
```tsx
  const uniqueAgreementIds = new Set([...computeJobs.map(e => e.agreementId)])
  const uniqueComputeJobs = computeJobs.filter(e => !uniqueAgreementIds.has(e.agreementId))
  const sortedComputeJobs = uniqueComputeJobs.sort((a,b) =>
    // ...
  )

  return sortedComputeJobs
```

Reply:
Creating the set before filtering will not work because the items would be all incorrectly detected as "duplicates". But I can split the filter from the sort if you think it's more readable